test/e2e/branding_test.go

```diff
@@ -0,0 +1,81 @@
+package e2e
+
+import (
+	"k8s.io/apimachinery/pkg/util/wait"
+	"testing"
+	"time"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/testframework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Test the customization features of the config map
+
+// the simpler test first would be branding set via the operator config
+func TestOperatorConfigBranding(t *testing.T) {
+	t.Logf("Start branding ConfigMap")
+	online := operatorsv1.BrandDedicated
+	client := testframework.MustNewClientset(t, nil)
+	// defer the cleanup
+	// we should prob have a Reset() func, to just put the operator config back to defaults
+	defer testframework.Reset(t, client, operatorsv1.Normal)
+	// helper to update the operator config
+	err := testframework.SetBrand(t, client, online)
+	if err != nil {
+		t.Fatalf("error: %s", err)
+	}
+	// now check the console-config in openshift-console & verify it has brand set
+	gotBrand := testframework.GetBrand(t, client)
+	if !brandsEqual(online, gotBrand) {
+		t.Fatalf("error: brand expected %v but brand got was %v", online, gotBrand)
+	}
+}
+
+func TestBrandingFromManagedConfigMap(t *testing.T) {
+	client := testframework.MustNewClientset(t, nil)
+
+	t.Logf("merging Data on the console ConfigMap")
+	expectedBranding := "dedicated"
+
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "console-config",
+			Namespace: "openshift-config-managed",
+		},
+		Data: map[string]string{
+			"console-config.yaml": `kind: ConsoleConfig
+apiVersion: console.openshift.io/v1
+customization:
+  branding: ` + expectedBranding + `
+  documentationBaseURL: https://docs.okd.io/4.0/
+`,
+		},
+	}
+
+	_, err := client.ConfigMaps(consoleapi.OpenshiftConfigManagedNamespace).Update(cm)
```

Comment:
Perhaps extract out the "create or update" logic here into a func.  We use `resourceapply.ApplyConfigMap()` in the actual operator code (see `SyncConfigMap()`), which you could also just use for this (though it requires an event recorder).

Reply:
yeah... that code was not meant to stay in there